.travis_do

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+set -e
+
+if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then make -C ${TRAVIS_BUILD_DIR}/ad9361/sw/ -f Makefile.linux ; fi
+make -C ${TRAVIS_BUILD_DIR}/ad9361/sw/ -f Makefile.generic
+cppcheck --quiet --force --error-exitcode=1 .
+
+COMMIT_RANGE=$TRAVIS_COMMIT_RANGE
+if [ -z "$TRAVIS_PULL_REQUEST_SHA" ]
+then
+	echo "Using only latest commit, since there is no Pull Request"
+	COMMIT_RANGE=HEAD~1
+fi
+
+FILES=$(git diff --name-only $COMMIT_RANGE)
+
+is_source_file() {
+	local file="$1"
+	local ext_list=".c .h"
+
+	for ext in $ext_list; do
+		[[ ${file: -2} == $ext ]] && return 0
```

Comment:
it's a good idea to use quotes in shell where possible; otherwise this statement can fail if ` ${file: -2}` expands to empty

Reply:
i.e. `[ "${file: -2}" == "$ext" ]`   ; 
single brackets can be used here too